lib/pbench/cli/agent/commands/tools/clear.py

```diff
@@ -26,88 +27,20 @@ def execute(self) -> int:
-                    # empty tool directories lingering or wasn't cleaned before.
-                    try:
-                        if group != "default":
-                            shutil.rmtree(self.tool_group_dir)
-                    except OSError:
-                        self.logger.error(
-                            "Failed to remove empty group directory %s",
-                            self.tool_group_dir,
-                        )
-                        return 1
-
-            tools_not_found = []  # Not found, yet
-            for remote in remotes:
-                tg_dir_r = self.tool_group_dir / remote
-                if not tg_dir_r.exists():
-                    self.logger.warn(
-                        'No remote host "%s" in group %s.', remote, group,
-                    )
-                    continue
-
-                if self.context.name:
-                    names = self.context.name.split(",")
-                else:
-                    # Discover all the tools registered for this remote
-                    names = self.tools(tg_dir_r)
-                    if not names:
-                        # FIXME:  this is another odd case -- the remote subdirectory
-                        #  exists, but it's empty.  (We'll remove it below.)
-                        self.logger.warn(
-                            'No tools in group "%s" on host "%s".', group, remote,
-                        )
-
-                for name in names:
-                    status = self._clear_tools(name, remote, group)
-                    if status > 0:
-                        errors = 1
-                        tools_not_found.append(name)
-                    elif status < 0:
-                        tools_not_found.append(name)
-
-                tool_files = [p.name for p in tg_dir_r.iterdir()]
-
-                if len(tool_files) == 1 and tool_files[0] == "__label__":
-                    label = tg_dir_r / "__label__"
-                    try:
-                        label.unlink()
-                    except Exception:
-                        self.logger.error(
-                            "Failed to remove label for remote %s", tg_dir_r
-                        )
-                        errors = 1
-
-                if self.is_empty(tg_dir_r):
-                    self.logger.info(
-                        'All tools removed from group "%s" on host "%s"',
-                        group,
-                        tg_dir_r.name,
-                    )
-                    try:
-                        shutil.rmtree(tg_dir_r)
-                    except OSError:
-                        self.logger.error(
-                            "Failed to remove remote directory %s", tg_dir_r
-                        )
-                        errors = 1
+            errors, tools_not_found = self._clear_remotes(group)
 
             if self.context.name and tools_not_found:
```

Comment:
I suppose that there's no harm in it, but I'm wondering if we can omit the check of `self.context.name` and just go based on the contents of `tools_not_found`.  (Previously, we needed to check whether the name was specified because we might try to print it out; now, we have an authoritative list which we'll print instead....)